src/cogs/unquote_link.py

```diff
@@ -0,0 +1,36 @@
+import re
+import urllib.parse
+
+from discord.ext import commands
+
+URL_PATTERN = re.compile(r"http(s)?://([-\w]+\.)+[-\w]+(/[-\w./?%&=~]*)?")
+
+
+class UnquoteLink(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        if message.author.bot:
+            return
+        if URL_PATTERN.search(message.content) is None:
+            return
+
+        urls = []
+        for url_match in URL_PATTERN.finditer(message.content):
+            url = url_match[0]
+            new_url = urllib.parse.unquote(url)
+            if new_url != url:
+                urls.append(new_url)
+
+        if len(urls) == 0:
+            return
+
+        new_message = "もしかして：\n" + "\n".join("<" + url + ">" for url in urls)
```

Comment:
```suggestion
        new_message = "もしかして：\n" + "\n".join(f"<{url}>" for url in urls)
```
f-strings のほうがかっこいいと思う！

Reply:
それパッと出てこなくてめんどくさくて、concatでも言うて見た目変わらんなって気分になってそれでごまかしちゃった。